lib/fd.js

```diff
@@ -0,0 +1,309 @@
+    if (this._writeStream) {
+      if (this._writeStream.destroyed) {
+        this._writeStream = null
+      } else {
+        return this._writeStream.end(err => {
+          if (err) return cb(err)  
+          this._writeStream = null
+          return cb(null)
+        })
+      }
+    }
+    if (this._range) {
+      this.drive.contentFeed.undownload(this._range)
+      this._range = null
+    }
+    process.nextTick(cb, null)
+  }
+
+  /**
+   * Will currently request the next 16 blocks linearly.
+   *
+   * TODO: This behavior should be more customizable in the future.
+   */
+  _refreshDownload (start, cb) {
+    let end = Math.min(this.stat.blocks + this.stat.offset, start + 16)
+
+    if (this._range) {
+      this.drive.contentFeed.undownload(this._range)
+    }
+
+    this._range = this.drive.contentFeed.download({ start, end, linear: true }, cb || noop)
+  }
+
+  _seekAndRead (buffer, offset, len, pos, cb) {
+    const start = this.stat.offset
+    const end = start + this.stat.blocks
+
+
+    this.drive.contentFeed.seek(this.stat.byteOffset + pos, { start, end }, (err, blk, blockOffset) => {
+      if (err) return cb(err)
+      this.position = pos
+      this.blockPosition = blk
+      this.blockOffset = blockOffset
+
+      this._refreshDownload(blk)
+      this._read(buffer, offset, len, cb)
+    })
+  }
+
+  _read (buffer, offset, len, cb) {
+    const self = this
+
+    let totalRead = 0
+    let available = len
+    readNextBlock()
+
+    function readNextBlock () {
+      self._readBlock(buffer, offset + totalRead, Math.max(available, 0), (err, bytesRead) => {
+        if (err) return cb(err)
+        if (!bytesRead) return cb(null, totalRead, buffer)
+
+        totalRead += bytesRead
+        available -= bytesRead
+
+        if (available > 0) {
+          return readNextBlock()
+        }
+        return cb(null, totalRead, buffer)
+      })
+    }
+  }
+
+  _readBlock (buffer, offset, len, cb) {
+    const buf = buffer.slice(offset, offset + len)
+    const blkOffset = this.blockOffset
+    const blk = this.blockPosition
+
+    if (this._range && (blk < this._range.start || blk > this._range.end)) {
+      this._refreshDownload(blk)
+    }
+
+    if ((this.stat.offset + this.stat.blocks) <= blk || blk < this.stat.offset) {
+      return process.nextTick(cb, null, 0, buffer)
+    }
+
+    this.drive.contentFeed.get(blk, (err, data) => {
+      if (err) return cb(err)
+      if (blkOffset) data = data.slice(blkOffset)
+
+      data.copy(buf)
+      const read = Math.min(data.length, buf.length)
+
+      if (blk === this.blockPosition && blkOffset === this.blockOffset) {
+        this.position += read
+        if (read === data.length) {
+          this.blockPosition++
+          this.blockOffset = 0
+        } else {
+          this.blockOffset = blkOffset + read
+        }
+      }
+
+      cb(null, read, buffer)
+    })
+  }
+}
+
+FileDescriptor.create = function (drive, name, flags, cb) {
```

Comment:
If I'm not mistaken this is the only func we actually use outside this module? If so we should module.exports it instead of the class.

Reply:
Yep, updated to export only this function